### PR TITLE
remove self.training check for VFNet head

### PR DESCRIPTION
**Pull request by dnth**
## Motivation

Self.training check causes error when model is validation mode because of inconsistent return values.

I find that the `forward_single` function returns three values `cls_score, bbox_pred, bbox_pred_refine` during training and returns only two values `cls_score, bbox_pred_refine` when it is not in training eg. validation.

## Modification

Remove the self.training check in the `forward_single` function.

Closes

https://github.com/open-mmlab/mmdetection/issues/6851

and 
https://github.com/open-mmlab/mmdetection/issues/7167

### CODE REVIEW

**Comment by BIGWangYuDong** [comment]
If we get three parameters during testing, this may cause an error?

During training, we will use `forward_train` in BaseDenseHead to get the result of VFNetHead outputs,
https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/dense_heads/base_dense_head.py#L330-L336
then calculate loss in VFNetHead, which will use three parameters `cls_score, bbox_pred, bbox_pred_refine`,
https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/dense_heads/vfnet_head.py#L351-L359

However, during testing, we will use `simple_test_bboxes` in  BBoxTestMixin
https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/dense_heads/dense_test_mixins.py#L36-L39
which will use `get_bboxes` to get results, but seems that we will not use three parameters?
https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/dense_heads/base_dense_head.py#L32-L41

**Comment by dnth** [comment]
I see.. i have overlooked testing.. is there any solution you think can handle the potential issue during testing?

**Comment by BIGWangYuDong** [comment]
> I see.. i have overlooked testing.. is there any solution you think can handle the potential issue during testing?

The potential issue may cause gt_label is None?

https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/dense_heads/base_dense_head.py#L330-L336

**Comment by dnth** [comment]
I ran `python tools/test.py configs/vfnet/vfnet_r50_fpn_1x_coco.py vfnet_r50_fpn_1x_coco_20201027-38db6f58.pth --show`
and somehow it does not raise any errors. Can anyone else verify?

**Comment by BIGWangYuDong** [comment]
> I ran `python tools/test.py configs/vfnet/vfnet_r50_fpn_1x_coco.py vfnet_r50_fpn_1x_coco_20201027-38db6f58.pth --show` and somehow it does not raise any errors. Can anyone else verify? The detections are also intact.

I'll have a check

**Comment by BIGWangYuDong** [comment]
before change the code:
企业微信截图_16450701491830

after change the code:
企业微信截图_16450704021799

**Comment by BIGWangYuDong** [comment]
@dnth Thanks for your PR, if you checked the reason for the potential bug, please feel free to re-open this PR or create a new PR. I'll close this PR now.

**Comment by dnth** [comment]
@BIGWangYuDong can we re-consider this PR? Is there a community discussion ongoing?

**Comment by OpenMMLab-Assistant003** [comment]
Hi @dnth！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @dnth
